src/main/java/com/teragrep/rlp_03/context/buffer/BufferLeaseImpl.java

```diff
@@ -51,55 +47,27 @@ public ByteBuffer buffer() {
 
     @Override
     public void addRef() {
-        lock.lock();
-        try {
-            refCount++;
-        }
-        finally {
-            lock.unlock();
-        }
+        phaser.register();
     }
 
     @Override
     public void removeRef() {
-        lock.lock();
-        try {
-
-            long newRefs = refCount - 1;
-            if (newRefs < 0) {
-                throw new IllegalStateException("refs must not be negative");
-            }
-
-            refCount = newRefs;
-        }
-        finally {
-            lock.unlock();
-        }
+        phaser.arriveAndDeregister();
     }
 
     @Override
     public boolean isRefCountZero() {
```

Comment:
this must be changed so that it checks if the phaser isTerminated

Reply:
fix 334d130d91728b1a8209c4d43affee26ed4c2baa